x-pack/platform/plugins/private/reporting/server/config/index.test.ts

```diff
@@ -36,20 +36,16 @@ const applyReportingDeprecations = (settings: Record<string, any> = {}) => {
 };
 
 describe('deprecations', () => {
-  it('logs a warning of roles.enabled for defaults', () => {
+  it('does not log a warning of roles.enabled for defaults', () => {
     const { messages } = applyReportingDeprecations({});
-    expect(messages).toMatchInlineSnapshot(`
-      Array [
-        "The default mechanism for Reporting privileges will work differently in future versions, which will affect the behavior of this cluster. Set \\"xpack.reporting.roles.enabled\\" to \\"false\\" to adopt the future behavior before upgrading.",
-      ]
-    `);
+    expect(messages).toMatchInlineSnapshot(`Array []`);
   });
 
-  it('logs a warning if roles.enabled: true is set', () => {
-    const { messages } = applyReportingDeprecations({ roles: { enabled: true } });
+  it('logs a warning if roles.allow: true is set', () => {
+    const { messages } = applyReportingDeprecations({ roles: { allow: true } });
```

Comment:
`roles.allow` takes an array of role names which would be allowed to generate reports, so I would change this to:
```suggestion
  it(`logs a warning if roles.allow: ['my_reporting_role'] is set`, () => {
    const { messages } = applyReportingDeprecations({ roles: { allow: ['my_reporting_role'] } });
```

Fortunately, it looks like the test passes either way :)

Reply:
Which makes this catch even more valuable - thank you!